src/main/java/EntryPoint/controller/AuthController.java

```diff
@@ -0,0 +1,108 @@
+package EntryPoint.controller;
+
+import EntryPoint.dto.ResponseDTO;
+import EntryPoint.dto.UserDTO;
+import EntryPoint.service.UserService;
+import EntryPoint.utils.JwtUtil;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+public class AuthController {
+    private final UserService userService;
+    private final JwtUtil jwtUtil;
+    ResponseDTO response;
+
+    @PostMapping("/register")
+    public ResponseEntity<ResponseDTO> register(@Valid @RequestBody UserDTO request) {
+        try {
+            Object registeredEmail = userService.register(request);
+            response = new ResponseDTO(true, "Registration Successful", registeredEmail);
+            return ResponseEntity.ok(response); // Return success response with email as data
+        } catch (Exception e) {
```

Comment:
Please catch exception if already a user is exists.

Reply:
resolved